lib/actions.js

```diff
@@ -11,12 +11,48 @@ var fs = require('fs');
 var keys = Object.keys;
 
 /**
+ * Helper function to query an element by XPath.
+ *
+ * @param {String} xpath
+ */
+
+function xpathLookup(xpath) {
+    try {
+        return document.evaluate(xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE).singleValueNode;
+    } catch (e) {
+        if (e instanceof DOMException) {
+            return undefined;
+        } else {
+            throw e;
+        }
+    }
+}
+
+/**
+ * Helper function to query an element by CSS selector.
+ *
+ * @param {String} selector
+ */
+
+function cssLookup(selector) {
+    try {
+        return document.querySelector(selector);
+    } catch (e) {
+        if (e instanceof DOMException) {
+            return undefined;
+        } else {
+            throw e;
+        }
+    }
+}
+
+/**
  * Get the version info for Nightmare, Electron and Chromium.
  * @param {Function} done
  */
-exports.engineVersions = function(done){
-  debug('.engineVersions()');
-  done(null, this.engineVersions);
+exports.engineVersions = function(done) {
+    debug('.engineVersions()');
```

Comment:
You should probably adjust your code editor to be two spaces and undo all of the whitespace formatting changes in this file.